AWSPinpoint/AWSPinpointEvent.m

```diff
@@ -160,11 +155,11 @@ - (void)addMetric:(NSNumber *)theValue forKey:(NSString *)theKey {
     }
     
     @synchronized(self.metrics) {
-        if(self.currentNumOfAttributesAndMetrics < MAX_NUM_OF_METRICS_AND_ATTRIBUTES) {
+        if(self.metrics.count < MAX_NUM_OF_METRICS_AND_ATTRIBUTES) {
```

Comment:
I think the correct quota check would be the same as before BUT the comparison should be <=. Also the same warning applies.